### ISSUE TITLE
Query - is the 'An Arbitrary ERC Model' used in framework a pre-trained model?

### ISSUE THREAD

**Issue by SupriyaUpadhyaya** [TaoShi1998/SSLCL#2]
Hello @TaoShi1998,

We are trying to implement the proposed framework using a transformer based model as 'An Arbitrary ERC Model'([figure 2](https://arxiv.org/pdf/2310.16676.pdf)). We are jointly training 'ERC Model' and the 'Label Learning' model. 'ERC Model' is trained from scratch. We are not using a pre-trained model. 

Please could you confirm if above implementation is in line with the [proposed framework](https://arxiv.org/pdf/2310.16676.pdf). Or do we have to use a pre-trained  ERC model as 'An Arbitrary ERC Model' in the framework?

Thanks and regards,
Supriya

**Comment by TaoShi1998** [comment]
Hi Supriya, thank you for your interest in our work. As for your question, the "arbitrary ERC model" means that there is no restriction for ERC model, it can be non pre-trained just like the DialogueRNN and DialogueCRN we used for comparison in the paper. So the answer is yes, you can use a non pre-trained model to implement the framework.

**Comment by SupriyaUpadhyaya** [comment]
Hello @TaoShi1998,

Thanks for confirmation that we could use either pre-trained or non pre-trained network for 'ERC model'.

Additionally, requesting your to please confirm on below queries:

1. We are training both 'ERC model' and 'Label learning' model using back propagation. We are calculating the SSLCL + cross entropy loss for ERC model and back propagating the same through both ERC network and label learning network using two different optimizers in PyTorch  (Code screenshot attached). Please can you confirm this training approach is correct as per the framework.

2. If we use a non pre-trained ERC model then the parameters of the ERC model are to be optimized based on SSLCL + cross entropy loss. The SSLCL loss is dependent on the label embeddings. The label embeddings inturn need to be optimized based on the SSLCL + cross entropy loss from ERC model(which is also not yet optimized). With such interdependency, how is it guaranteed that the optimal label embeddings are learned ?

<img width="1021" alt="sslcl-joint-training" src="https://github.com/TaoShi1998/SSLCL/assets/18483906/cd64d700-c584-4732-aaec-fc8180381208">


Thanks and regards,
Supriya
